src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -69,6 +69,12 @@ private void AttachedPeer_MessageReceived(NetworkPeer peer, IncomingMessage mess
                     var addressPayload = new AddrPayload(endPoints.Select(p => new NetworkAddress(p)).ToArray());
                     peer.SendMessageVoidAsync(addressPayload);
                 }
+
```

Comment:
We should probably do this for both ping and pong no?

Also note `this.AttachedPeer` might be null, its safer to create a local variable.